hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/TestAbfsInputStream.java

```diff
@@ -524,30 +527,33 @@ public void testStreamPurgeDuringReadAheadCallExecuting() throws Exception {
     final ReadBufferManager readBufferManager
         = ReadBufferManager.getBufferManager();
 
+    final int readBufferTotal = readBufferManager.getNumBuffers();
+
     //Sleeping to give ReadBufferWorker to pick the readBuffers for processing.
-    Thread.sleep(1_000L);
+    Thread.sleep(readBufferTransferToInProgressProbableTime);
 
     Assertions.assertThat(readBufferManager.getInProgressCopiedList())
-        .describedAs("InProgressList should have 3 elements")
-        .hasSize(3);
+        .describedAs("InProgressList should have " + readBufferQueuedCount + " elements")
+        .hasSize(readBufferQueuedCount);
+    final int freeListBufferCount = readBufferTotal - readBufferQueuedCount;
     Assertions.assertThat(readBufferManager.getFreeListCopy())
-        .describedAs("FreeList should have 13 elements")
-        .hasSize(13);
+        .describedAs("FreeList should have " + freeListBufferCount + "elements")
```

Comment:
you can actually use string.format patterns here; most relevant for on demand toString calls which are more expensive. I'm not worrying about it here though